src/test/java/org/folio/innreach/controller/InnReachTransactionControllerTest.java

```diff
@@ -1954,6 +1966,65 @@ void return400_when_transferLocalHoldItem_and_transactionIsNotOfLocalHold() {
     assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
   }
 
+  @Test
+  @Sql(scripts = {
+    "classpath:db/inn-reach-recall-user/pre-populate-inn-reach-recall-user.sql",
+    "classpath:db/central-server/pre-populate-central-server-with-recall-user.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql"
+  })
+  void recallItemHoldWhenRequestStatusOpenNotYetFilled() {
+    var transaction = repository.fetchOneById(UUID.fromString(TRANSACTION_WITH_ITEM_HOLD_ID)).get();
+    transaction.setState(ITEM_RECEIVED);
+    repository.save(transaction);
+
+    var loan = new LoanDTO();
+    var currentDate = new Date();
+    loan.setDueDate(currentDate);
+    var intCurrentDate = (int) currentDate.toInstant().truncatedTo(ChronoUnit.SECONDS).getEpochSecond();
+
+    when(circulationClient.queryRequestsByItemId(PRE_POPULATED_FOLIO_ITEM_ID)).thenReturn(getOpenRequests());
+    when(circulationClient.findLoan(PRE_POPULATED_FOLIO_LOAN_ID)).thenReturn(Optional.of(loan));
+    when(innReachClient.postInnReachApi(any(), anyString(), anyString(), anyString(), any())).thenReturn("response");
+
+    var responseEntity = testRestTemplate.postForEntity(
+      ITEM_HOLD_RECALL_ENDPOINT, null, Void.class, TRANSACTION_WITH_ITEM_HOLD_ID);
+
+    var transactionAfterRecall = repository.fetchOneById(UUID.fromString(TRANSACTION_WITH_ITEM_HOLD_ID)).get();
+
+    assertEquals(HttpStatus.NO_CONTENT, responseEntity.getStatusCode());
+    assertEquals(InnReachTransaction.TransactionState.RECALL, transactionAfterRecall.getState());
+    assertEquals(intCurrentDate, transactionAfterRecall.getHold().getDueDateTime());
+  }
+
+
+  @Test
+  @Sql(scripts = {
+    "classpath:db/inn-reach-recall-user/pre-populate-inn-reach-recall-user.sql",
+    "classpath:db/central-server/pre-populate-central-server-with-recall-user.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql",
+  })
+  void recallItemHoldWhenRequestStatusNotOpen() {
+
+    var transaction = repository.fetchOneById(UUID.fromString(TRANSACTION_WITH_ITEM_HOLD_ID)).get();
+    transaction.setState(ITEM_RECEIVED);
+    repository.save(transaction);
+
+    when(circulationClient.queryRequestsByItemId(PRE_POPULATED_FOLIO_ITEM_ID)).thenReturn(getNotOpenRequests());
+
+    var responseEntity = testRestTemplate.postForEntity(
+      ITEM_HOLD_RECALL_ENDPOINT, null, Void.class, UUID.fromString(TRANSACTION_WITH_ITEM_HOLD_ID));
+
+    assertEquals(HttpStatus.NO_CONTENT, responseEntity.getStatusCode());
+
+    var centralServerCodeCaptor = ArgumentCaptor.forClass(String.class);
+    verify(recallUserService).getRecallUserForCentralServer(centralServerCodeCaptor.capture());
+
+    var centralServerCode = centralServerCodeCaptor.getValue();
+
+    assertEquals("d2ir", centralServerCode);
+  }
```

Comment:
how does this test verify that recall request was sent?

Reply:
Done